cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java

```diff
@@ -1,39 +1,35 @@
 package com.epam.cleancode.comments;
 
-public class MortgageInstallmentCalculator {
+
+class MortgageInstallmentCalculator {
+
+    private static final String INVALID_INPUT_EXCEPTION = "Negative values are not allowed";
+    private static final double NUMBER_FOR_CONVERT_TO_DECIMAL = 100.0;
+    private static final double AMOUNT_OF_MONTHS = 12.0;
+    private static final double MIN_VALUE = 12.0;
 
     /**
      *
-     * @param p principal amount
-     * @param t term of mortgage in years
-     * @param r rate of interest
+     * @param loanAmount principal amount
+     * @param termDurationInYears term of mortgage in years
+     * @param digitaRateOfInterest rate of interest
      * @return monthly payment amount
      */
-    public static double calculateMonthlyPayment(
-            int p, int t, double r) {
-
-        //cannot have negative loanAmount, term duration and rate of interest
-        if (p < 0 || t <= 0 || r < 0) {
-            throw new InvalidInputException("Negative values are not allowed");
-        }
-
-        // Convert interest rate into a decimal - eg. 6.5% = 0.065
-        r /= 100.0;
+    static double calculateMonthlyPayment(final int loanAmount, final int termDurationInYears, double digitaRateOfInterest) {
 
-        // convert term in years to term in months
-        double tim = t * 12;
+        digitaRateOfInterest = digitaRateOfInterest / NUMBER_FOR_CONVERT_TO_DECIMAL;
```

Comment:
there is a common rule where all method's params have final modifier and it means that you can't reassign these values. Also each operation has outcome, this outcome should be named properly, from this operation I can't see what the result of this line. You use right way in line 22 but here it is invalid way :)